src/Tools/Source/CompilerGeneratorTools/Source/IOperationGenerator/IOperationClassWriter.cs

```diff
@@ -298,11 +319,222 @@ void writeEnumElement(string kind, int value, string operationName, string? extr
+                        Write($"{prop.Type} {prop.Name.ToCamelCase()}, ");
+                    }
+
+                    writeStandardConstructorParameters(type.IsAbstract);
+                    WriteLine(")");
+                    Indent();
+                    Write(": base(");
+
+                    var @base = _typeMap[type.Base];
+                    if (@base is object)
+                    {
+                        foreach (var prop in GetAllProperties(@base).Where(p => !IsIOperationType(p.Type)))
+                        {
+                            Write($"{prop.Name.ToCamelCase()}, ");
+                        }
+                    }
+
+                    writeStandardBaseArguments(type switch
+                    {
+                        { IsAbstract: true } => "kind",
+                        { IsInternal: true } => "OperationKind.None",
+                        _ => $"OperationKind.{GetKind(type)}"
+                    });
+                    Outdent();
+                    // The base class is responsible for initializing its own properties,
+                    // so we only initialize our own properties here.
+                    if (type.Properties.Count == 0)
+                    {
+                        // Note: our formatting style is a space here
+                        WriteLine(" { }");
+                    }
+                    else
+                    {
+                        Blank();
+                        Brace();
+                        foreach (var prop in type.Properties)
+                        {
+                            WriteLine($"{prop.Name} = {prop.Name.ToCamelCase()};");
+                        }
+                        Unbrace();
+                    }
+
+                    foreach (var prop in type.Properties)
+                    {
+                        WriteLine($"public {prop.Type} {prop.Name} {{ get; }}");
+                    }
+
+                    if (type is Node node)
+                    {
+
+                        var visitorName = GetVisitorName(node);
+                        WriteLine($@"public override IEnumerable<IOperation> Children => Array.Empty<IOperation>();
+        public override void Accept(OperationVisitor visitor) => visitor.{visitorName}(this);
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument) => visitor.{visitorName}(this, argument);");
+                    }
 
-        private bool IsImmutableArray(string typeName)
+                    Unbrace();
+                }
+            }
+
+            WriteLine("#endregion");
+
+            void writeStandardConstructorParameters(bool isAbstract)
+            {
+                if (isAbstract)
+                {
+                    Write("OperationKind kind, ");
+                }
+
+                Write("SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit");
```

Comment:
this isn't that important as long as the generated code is correct, but it seems weird to include the trailing paren in writeStandardBaseArguments but not here.